citrination_client/views/data_view_builder.py

```diff
@@ -58,6 +61,69 @@ def add_raw_descriptor(self, descriptor):
         """
         self.configuration['descriptors'].append(descriptor)
 
+    def add_relation(self, inputs, output, relation_type='lolo'):
+        """
+        Add a manual relation.  If no relations are manually added, Citrination will automatically generate
+        relations when the view is created.
+
+        :param inputs: Array of strings or a single string of descriptor key(s) for the relation inputs
+        :param output: Single string descriptor key for the relation output
+        :param relation_type: Kind of relation, currently only 'lolo' is supported
+        """
+        relation_obj = {}
+
+        if isinstance(inputs, str):
+            relation_obj['inputs'] = [inputs]
+        elif isinstance(inputs, list):
+            if len(inputs) == 0:
+                raise ValueError("Inputs list must not be empty")
+            relation_obj['inputs'] = inputs
+        else:
+            raise ValueError("Unexpected type for inputs, expecting either a string or list of strings")
+
+        if isinstance(output, str):
+            relation_obj['output'] = output
```

Comment:
Do we need to check if the output is equal to the input?

Reply:
Added this check also.